lib/ThreeScaleClient.php

```diff
@@ -279,11 +339,55 @@ public function authrep($appId, $appKey = null, $usage = null, $userId = null, $
     
   }
 
-  public function authrep_with_user_key($userKey, $usage = null, $userId = null, $object = null, $no_body = null, $serviceId = null) {  
+  /**
+   * Authorize and report in a single shot for user key auth mode
+   *
+   * @param $userKey  User key.
+   * @param $serviceId service id.
+   *
+   * @return ThreeScaleResponse object containing additional authorization information.
+   * If both provider key and application id are valid, the returned object is actually
+   * @see ThreeScaleAuthorizeResponse (which is derived from ThreeScaleResponse) and
+   * contains additional information about the usage status.
+   *
+   * @see ThreeScaleResponse
+   * @see ThreeScaleAuthorizeResponse
+   *
+   * @throws ThreeScaleServerError in case of unexpected internal server error
+   *
+   * Example:
+   *
+   * <code>
+   *   <?php
+   *   $response = $client->authorize('user_key');
+   *   // or $response = $client->authorize('user_key','service_id');
+   *
+   *   if ($response->isSuccess()) {
+   *     // ok.
+   *   } else {
+   *     // something is wrong.
+   *   }
+   *   ?>
+   * </code>
+   */
+
+  public function authrep_with_user_key($userKey, $serviceId = null, $usage = null, $userId = null, $object = null, $no_body = null) {  
```

Comment:
The same here. Reordering parameters looks like a breaking change that can break every app that upgrades. Not sure if PHP supports keyword arguments but this would be a good use case.